stairway/src/main/java/bio/terra/stairway/QueueInterface.java

```diff
@@ -0,0 +1,32 @@
+package bio.terra.stairway;
+
+public interface QueueInterface {
+
+  /**
+   * Read some messages from the queue and request that Stairway process each one.
+   *
+   * @param dispatchContext anonymous context passed to the processFunction
+   * @param maxMessages Maximum messages to attempt to read at once. Fewer messages may be read.
+   * @param processFunction Function to call for each message. The function returns true if the
+   *     message is successfully handled; false if it was not and should remain on the queue.
+   * @throws InterruptedException allows for queue waits to throw if the calling thread is shutdown
+   */
+  void dispatchMessages(
+      Object dispatchContext, int maxMessages, QueueProcessFunction processFunction)
+      throws InterruptedException;
+
+  /**
+   * Put a message on the queue.
+   *
+   * @param message the message to enqueue
+   * @throws InterruptedException allows for queue waits to throw if the calling thread is shutdown
+   */
+  void enqueueMessage(String message) throws InterruptedException;
```

Comment:
The reason I cannot do a simple `dequeue` interface is the requirement to ack or nack the message based on successful processing. That is the method PubSub has so that we avoid dropping messages on the floor in event of failure. They are not removed from the queue until they are ack'd. In this specific case, processing results in a successful database transaction. Stairway is set up to tolerate receiving the same message twice, but it will not tolerate losing a message in the queue.